### ISSUE TITLE
destructuring object with getters results in incorrect result

### ISSUE THREAD

**Issue by coderkd10** [airbnb/javascript#1655]
[Rule 5.1 object destructuring](https://github.com/airbnb/javascript#destructuring--object) recommends that we use destructing instead of accessing multiple properties of an object, and there is an eslint rule that enforces this.

But incase the object contains getters we cannot destructure it, and have to resort to accessing its properties manually.

Example - 

```js
const boundingRect = document.body.getBoundingClientRect();

// Incorrect way
const {
    x,
    y,
    height,
    width
} = boundingRect;
// x, y, height, width are undefined


// correct way
const x = boundingRect.x;
const y = boundingRect.y;
//...
```

**Comment by ljharb** [comment]
So, two things:
 1. This guide forbids using getters and setters, so you wouldn't run into this with your own objects, only with built-in things
 1. Object destructuring uses `[[Get]]`, as does member access, so it works identically the same.

If `const x = boundingRect.x` works, then `const { x } = boundingRect` absolutely works, 100% - that's part of the JS specification.

**Comment by coderkd10** [comment]
@ljharb thanks! you are correct. The example that I posted works correctly as expected and `x`, `y`, are not actually undefined.

Actually I was confused since the code that originally wrote was some thing like

```
let x;
const initialize = () => {
    let { x } = document.body.getBoundingClientRect();  
}
initialize();
console.log(x);
```
And I was (wrongly) expecting to see `x` be assigned some value at the end, and when I saw it be undefined I again wrongly concluded that it was due to object destructing (maybe I was drunk that evening!). The correct reasoning is pretty obvious now - a new local variable with name `x` was created inside the function, and it overshadowed the `x` in the parent scope; when then function returned the local variables when out of scope and were cleared; and the variable `x` in the parent scope was never touched and hence had to be undefined.

Thanks for help again!
